textalyze.js

```diff
@@ -1,23 +1,32 @@
-/**
- * This is the base code for v0.1 of our JavaScript text analyzer.
- * Visit https://github.com/jfarmer/text-analysis to see what to do.
- *
- * Send a message in Slack if you're stuck or unsure what to do.  These
- * comments are here to help you, but please delete them as you go along.
- */
-
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  let counts = new Map();
-
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  let counts = array.reduce((accum, arrVal) => {
+    var newCount = accum.has(arrVal) ? accum.get(arrVal) + 1 : 1;
+    return accum.set(arrVal, newCount);
+  }, new Map());
 
   return counts;
 }
 
-module.exports = { itemCounts };
+function stringToCharacters(input) {
```

Comment:
There are three ways to handle unexpected input:

1. Assert — throw an error
1. Coerce — convert the input into something you can handle
1. Ignore — do nothing and let something further down handle the error

In this case, you're coercing.  I'd say that if you want to coerce someone is more likely to expect you to convert the input to a String.  For example, it's surprising that the following are all `true`:

```
sanitize(12345) !== sanitize('12345');
sanitize(12345.toString()) === sanitize('12345');
sanitize(12345) === sanitize('')
```

I think the best course of action here is to ignore.